hw_4_flask.py

```diff
@@ -0,0 +1,82 @@
+import random
+
+
+import requests
+import pandas as pd
+
+from datetime import datetime
+from flask import Flask, jsonify, abort
+from faker import Faker
+from webargs import fields, validate
+from webargs.flaskparser import use_kwargs
+
+fake_instance = Faker('uk_UA')
+
+app = Flask(__name__)
+
+
+@app.route('/students')
+@use_kwargs(
+    {
+        'quantity': fields.Int(
+            load_default=50,
+            validate=[validate.Range(min=1, max=1000)]
+        )
+    },
+    location='query'
+)
+def generate_students(quantity):
+    students_data: list = [
+        {
+            'first_name': fake_instance.first_name(),
+            'last_name': fake_instance.last_name(),
+            'email': fake_instance.unique.email(),
+            'password': fake_instance.password(length=random.randint(10, 20)),
+            'birthday': str(fake_instance.date_of_birth(minimum_age=18, maximum_age=28))
+        } for data in range(quantity)
+    ]
+    df = pd.DataFrame(columns=['First Name', 'Last Name', 'Email', 'Password', 'Birthday'])
+    for p in students_data:
+        df = pd.concat([df, pd.DataFrame(p, index=[0])], ignore_index=True)
+        df.to_csv('students_data.csv', index=False)
```

Comment:
remove df.csv from for loop.

Using for with pandas is antipattern, don't do that))

https://stackoverflow.com/questions/20638006/convert-list-of-dictionaries-to-a-pandas-dataframe

Reply:
I made the code shorter. As I understand it, my mistake is that I for loop is an unnecessary action, as well as an additional indication of columns, since it will still take the name of the column from the key.

df = pd.DataFrame(students_data)
    df.to_csv('students_data.csv', index=False)